app/models/account_request.rb

```diff
@@ -40,28 +44,25 @@ def identity_token
     JWT.encode({ account_request_id: id }, Rails.application.secrets[:secret_key_base], 'HS256')
   end
 
+  # @return [Boolean]
   def confirmed?
-    confirmed_at.present?
+    pending? || approved?
   end
 
+  # @return [Boolean]
   def processed?
     organization.present?
   end
 
-  def approved?
-    "Approved" if confirmed? && Organization.present?
-  end
-
-  def pending_approval?
-    "Pending Approval" if confirmed? && Organization.blank?
-  end
-
-  def requested?
-    "Requested" unless confirmed?
+  def approve!
+    update!(confirmed_at: Time.current, status: 'pending')
+    AccountRequestMailer.approval_request(account_request_id: id).deliver_later
   end
 
-  def status
-    approved? || pending_approval? || requested?
+  # @param reason [String]
+  def reject!(reason)
+    update!(status: 'rejected', rejection_reason: reason)
+    AccountRequestMailer.rejection(account_request: self).deliver_later
```

Comment:
```suggestion
    AccountRequestMailer.rejection(account_request_id: self.id).deliver_later
```
I often use the `id` and make the `AccountRequestMailer` do the lookup for me. Delayed::Job has a weird issue where if you happen to (not likely at all!) delete the account_request before it gets processed then the worker node will fail (not just the job!) :O